swift-sdk/Internal/IterableRequest.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  Created by Tapash Majumder on 7/29/20.
+//  Copyright © 2020 Iterable. All rights reserved.
+//
+
+import Foundation
+
+// These are Iterable specific Request items.
+// They don't have Api endpoint and request endpoint defined yet.
+enum IterableRequest {
+    case get(GetRequest)
+    case post(PostRequest)
+}
+
+extension IterableRequest: Codable {
+    enum CodingKeys: String, CodingKey {
+        case type
+        case value
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let type = try container.decode(String.self, forKey: .type)
+        switch type.lowercased() {
+        case "get":
```

Comment:
Should "get" and "post" just be constants in our SDK? We could also remove the need to lower case in the switch.